Framework/PythonInterface/mantid/simpleapi.py

```diff
@@ -302,6 +302,39 @@ def fitting_algorithm(f):
     When applied to a function definition this decorator replaces its code with code of
     function 'wrapper' defined below.
     """
+    def gather_fitting_returns(function_name, lhs, algm):
```

Comment:
Why not declare the 'OutputNIterations' property in `Fit::createOutput()` after the other special output properties ('OutputNormalisedCovarianceMatrix', 'OutputParameters', etc.)? If I understood correctly, this would take care of the ordering automatically. Changing the ordering here looks to me, frankly speaking, like a hack.